### PR TITLE
Use Cache in ITC worker.

### PR DESCRIPTION
**Pull request by rpiaggio**
On the server side, restructure the logic a bit to use a `Cache`.

On the UI side, chunk responses to avoid intermediate rerenders.

### CODE REVIEW

**Review by toddburnside — APPROVED** [review]
LGTM

File: explore/src/main/scala/explore/config/SpectroscopyModesTable.scala [hunk 1]
```diff
@@ -499,24 +500,23 @@ object SpectroscopyModesTable {
...
-                                       // Update the cache
-                                       itcResults.modStateAsync(_.update(itcResponse))
-                                   )
-                                 )
+                    request <-
+                      ItcClient[IO]
+                        .request(ItcMessage.Query(w, sn, constraints, t, modes))
+                        .map(
+                          // Avoid intermediate rerenders. They are slow.
+                          _.groupWithin(100, 500.millis).evalTap(itcResponseChunk =>
```
Comment (cquiroz) [inline explore/src/main/scala/explore/config/SpectroscopyModesTable.scala:510]: What happens if you have less than 100 rows?

File: workers/src/main/scala/workers/ItcServer.scala [hunk 2]
```diff
@@ -36,8 +36,6 @@ import scala.scalajs.js
 
 import js.annotation._
 
-extension [A](a: AsyncCallback[A]) def toIO: IO[A] = asyncCallbackToIO.apply(a)
```
Comment (cquiroz) [inline workers/src/main/scala/workers/ItcServer.scala:39]: Not needed anymore?

File: explore/src/main/scala/explore/config/SpectroscopyModesTable.scala (hunk 1 above)
Comment (rpiaggio) [reply explore/src/main/scala/explore/config/SpectroscopyModesTable.scala:510]: The stream ends, and a chunk is emitted with just the rows that were received.

File: workers/src/main/scala/workers/ItcServer.scala (hunk 2 above)
Comment (rpiaggio) [reply workers/src/main/scala/workers/ItcServer.scala:39]: There's now a generic `toF[F]`

File: workers/src/main/scala/workers/itc/ITCRequests.scala [hunk 3]
```diff
@@ -94,14 +97,31 @@ object ITCRequests {
...
+                  .GenericError("Error calling ITC service"): ItcQueryProblems)
+                  .leftNec[ItcResult]
+              ).some.pure[F]
             )
         }
-        .getOrElse(Applicative[F].unit)
+        .sequence
+        .map(_.flatten)
+
+    val cacheableRequest = Cacheable(CacheName("itcQuery"), CacheVersion(1), doRequest)
```
Comment (cquiroz) [inline workers/src/main/scala/workers/itc/ITCRequests.scala:124]: I like the opaque types here

**Review by cquiroz — APPROVED** [review]
LGTM looking great

**Pull request merged**